### ISSUE TITLE
Improve Copilot Edits working set file limit handling 

### ISSUE THREAD

**Issue by bellgab** [microsoft/vscode-copilot-release#3181]
Type: <b>Bug</b>

While develop in  python, the chat window shows editing file, but nothin happens; the file didn't created/modified. The AI didn't realize that; the bug must be in VS Code plug-in.

Usualy it appears after 6-8 steps.

Extension version: 0.23.0
VS Code version: Code 1.96.0 (138f619c86f1199955d53b4166bef66ef252935c, 2024-12-11T02:29:09.626Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4800MQ CPU @ 2.70GHz (8 x 2694)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.43GB (7.38GB free)|
|Process Argv|--crash-reporter-id 4a5aa352-d05c-4e55-8ac6-9059dbbbcbe9|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805:30301674
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythonnoceb:30805159
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
dvdeprecation:31068756
dwnewjupytercf:31046870
nativerepl2:31139839
pythonrstrctxt:31112756
nativeloc1:31192215
cf971741:31144450
iacca1:31171482
notype1cf:31157160
5fd0e150:31155592
dwcopilot:31170013
stablechunks:31184530

```

</details>

**Comment by bellgab** [comment]
![Image](https://github.com/user-attachments/assets/5ef4a3d3-9eee-420e-b376-0d9a8a68181d)
[system.txt.txt](https://github.com/user-attachments/files/18115399/system.txt.txt)

Dear GitHub Team,

First of all, thank you for your hard work and support. I think GitHub Copilot is fantastic and has greatly facilitated my work.

Unfortunately, I can't reproduce the issue more specifically. It occurs on both Windows 10 Pro and Windows 11 Pro. The type of project and the language model do not seem to matter, as the issue always arises. Only restarting Windows seems to resolve it; restarting the plugins or VS Code does not help.

The attached screenshot and system info are from a Windows 11 environment, involving a Node.jsproject.

I should mention that this issue started occurring immediately after I switched to the Copilot Business plan and created an organization named "Bella-Gabor-EV". I added my user account and created a Development Team, with Copilot being automatically added as the first member.

Before this, I did not experience this issue. I mostly received messages that the communication supervisory subsystem blocked the response because it did not comply with the rules, asking me to rephrase my request.

Thank you in advance for your assistance!

Best regards, Gabor Bella Budapest, Hungary

**Comment by bellgab** [comment]
> ![Image](https://github.com/user-attachments/assets/5ef4a3d3-9eee-420e-b376-0d9a8a68181d) [system.txt.txt](https://github.com/user-attachments/files/18115399/system.txt.txt)
> 
> Dear GitHub Team,
> 
> First of all, thank you for your hard work and support. I think GitHub Copilot is fantastic and has greatly facilitated my work.
> 
> Unfortunately, I can't reproduce the issue more specifically. It occurs on both Windows 10 Pro and Windows 11 Pro. The type of project and the language model do not seem to matter, as the issue always arises. Only restarting Windows seems to resolve it; restarting the plugins or VS Code does not help.
> 
> The attached screenshot and system info are from a Windows 11 environment, involving a Node.jsproject.
> 
> I should mention that this issue started occurring immediately after I switched to the Copilot Business plan and created an organization named "Bella-Gabor-EV". I added my user account and created a Development Team, with Copilot being automatically added as the first member.
> 
> Before this, I did not experience this issue. I mostly received messages that the communication supervisory subsystem blocked the response because it did not comply with the rules, asking me to rephrase my request.
> 
> Thank you in advance for your assistance!
> 
> Best regards, Gabor Bella Budapest, Hungary

PS:
OK, I see it now, the files count in working set reachs th 10. When i remove the files, the Copilot can write again.. :)

**Comment by joyceerhl** [comment]
The working set is limited to 10 files. Can you please try removing some files from your working set before continuing with your editing session?

**Comment by bellgab** [comment]
Yes, i know it.
The problem is, the copilot automaticly create files, and give them to the work set.
When it fills, copilot continues its work, doen't realize, it can't  wirte anymore to files.
I can't delete so quickly...

I don't add files to the work set, whenever i can delete from it, copilot does some, 2-4 false modification.

________________________________
Feladó: Joyce Er ***@***.***>
Elküldve: 2024. december 21., szombat 17:07
Címzett: microsoft/vscode-copilot-release ***@***.***>
Másolatot kap: bellgab ***@***.***>; Author ***@***.***>
Tárgy: Re: [microsoft/vscode-copilot-release] Didnn't write code to file/editor (Issue #3181)


The working set is limited to 10 files. Can you please try removing some files from your working set before continuing with your editing session?

**Comment by joyceerhl** [comment]
> I don't add files to the work set, whenever i can delete from it, copilot does some, 2-4 false modification.

Do you mean that Copilot changes files that it shouldn't be changing? If so do you have steps to reproduce this behavior or a screen recording?

**Comment by bellgab** [comment]
Nope, the copilot correctly create files, and these files added to work set automaticly.

The problem: when the working set is full, the copilot continues the wrok, creating files, but the files didn't created; i must delete from working set, and ask copilot to repeat create the missing files.

________________________________
Feladó: Joyce Er ***@***.***>
Elküldve: 2024. december 21., szombat 20:11
Címzett: microsoft/vscode-copilot-release ***@***.***>
Másolatot kap: bellgab ***@***.***>; Author ***@***.***>
Tárgy: Re: [microsoft/vscode-copilot-release] Didnn't write code to file/editor (Issue #3181)


I don't add files to the work set, whenever i can delete from it, copilot does some, 2-4 false modification.

Do you mean that Copilot changes files that it shouldn't be changing? If so do you have steps to reproduce this behavior or a screen recording?

**Comment by joyceerhl** [comment]
Edits stops dealing with files once the working set is full (currently 10 files). There are at least two things we could do to improve this situation:
1. Raise the working set limit. This won't inherently improve the Edits experience because responses will just get truncated due to length reasons more often.
2. Do output chaining when the length limit is hit (i.e. feed the existing chat completion back into a new prompt so that the response continues). This is probably necessary for option 1 to lead to a meaningful UX improvement.
3. Don't enforce the working set limit when accepting text edits--since the model was able to produce them we should just take them (this is currently done [here](https://github.com/microsoft/vscode/blob/19e123108de47fcc4c3bf8efaca327dba2f4b2af/src/vs/workbench/contrib/chat/browser/chatEditing/chatEditingSession.ts#L632-L635) in VS Code)

cc @jrieken @aeschli @alexdima for thoughts

**Comment by bellgab** [comment]
Awesome, thanks for the update!

Joyce Er ***@***.***> ezt írta (időpont: 2025. febr. 18., K,
17:28):

**Comment by beingminimal** [comment]
> In latest Insiders we have lifted the working set file limit

So it will be unlimited and picked/dropped by a copilot automatically?

**Comment by joyceerhl** [comment]
Yes, that's correct @beingminimal

**Comment by beingminimal** [comment]
@joyceerhl 

Here is screen recording and image which shows copilot not working with unlimited files and not picking/dropping files based on prompt.

**Comment by joyceerhl** [comment]
Verification steps:
* Verify you can attach >10 files to a Copilot Edits request
* Make a few Edits requests and verify you do not hit a client side rate limit with guidance to retry after X seconds

**Comment by bellgab** [comment]
Thanls, it worked for me.

The new files inserted to the top, pushed out the oldest ones.

Joyce Er ***@***.***> ezt írta (időpont: 2025. febr. 25., K,
17:17):
